pyuvdata/uvcal/uvcal.py

```diff
@@ -1090,76 +1162,13 @@ def set_telescope_params(self, *, overwrite=False):
-                        telescope_ant_inds.append(
-                            np.where(telescope_obj.antenna_names == antname)[0][0]
-                        )
-                # next try using numbers
-                if len(ant_inds) != self.Nants_telescope:
-                    for index, antnum in enumerate(self.antenna_numbers):
-                        # only update if not already found
-                        if (
-                            index not in ant_inds
-                            and antnum in telescope_obj.antenna_numbers
-                        ):
-                            this_ant_ind = np.where(
-                                telescope_obj.antenna_numbers == antnum
-                            )[0][0]
-                            # make sure we don't already have this antenna associated
-                            # with another antenna
-                            if this_ant_ind not in telescope_ant_inds:
-                                ant_inds.append(index)
-                                telescope_ant_inds.append(this_ant_ind)
-                if len(ant_inds) != self.Nants_telescope:
-                    warnings.warn(
-                        "Not all antennas have positions in the telescope object. "
-                        "Not setting antenna_positions."
-                    )
-                else:
-                    params_set = ["antenna_positions"]
-                    if overwrite:
-                        self.antenna_names = telescope_obj.antenna_names
-                        self.antenna_numbers = telescope_obj.antenna_numbers
-                        self.antenna_positions = telescope_obj.antenna_positions
-                        self.Nants_telescope = telescope_obj.Nants_telescope
-                        params_set += [
-                            "antenna_names",
-                            "antenna_numbers",
-                            "Nants_telescope",
-                        ]
-                    else:
-                        telescope_ant_inds = np.array(telescope_ant_inds)
-                        self.antenna_positions = telescope_obj.antenna_positions[
-                            telescope_ant_inds, :
-                        ]
-                    params_set_str = ", ".join(params_set)
-                    warnings.warn(
-                        f"{params_set_str} are not set or are being "
-                        "overwritten. Using known values for "
-                        f"{telescope_obj.telescope_name}."
-                    )
-
-        else:
-            raise ValueError(
-                f"Telescope {self.telescope_name} is not in known_telescopes."
-            )
+        self.telescope.update_params_from_known_telescopes(
+            overwrite=overwrite,
+            warn=warn,
+            run_check=run_check,
+            check_extra=check_extra,
+            run_check_acceptability=run_check_acceptability,
+        )
```

Comment:
IMHO, as stated before, this `update` method should be removed and here we should be doing `self.telescope = Telescope.from_known_telescopes()`. But hold on -- you're not even passing any actual params to update here??

Reply:
It figures out what to update based on what is available for each telescope and what is already set on the object. This replaces old code that did the same.

Some known telescopes are just from Astropy sites, so all they have is a telescope location. Others have everything down to antenna positions. It's widely variable. And when we read in old cal files and UVFlag files there's a range of what metadata is set -- for very old files they are missing almost all the telescope metadata, while on new files this is basically a no-op because everything is already set.